hack/shellcheck.sh

```diff
@@ -28,6 +28,11 @@ else
     fi
 fi
 
+# Find and check shell script files with ShellCheck
 find . -name '*.sh' -not -path './vendor/*' -not -path './*/vendor/*' -not -path './git/*' \
     -not -path './bin/*' -not -path './testbin/*' -print0 \
     | xargs -0 --no-run-if-empty ${shellcheck} -x
+
+# All checks passed successfully
+echo "All checks passed successfully"
+exit 0
```

Comment:
You've discarded the exit status of the shellcheck command completely, so it will now always succeed.